### PR TITLE
enable hdfs support for windows

### PR DESCRIPTION
**Pull request by guschmue**

### CODE REVIEW

**Comment by caisq** [comment]
@tensorflow-jenkins test this please

**Comment by mrry** [comment]
Thanks for adding this, Guenther! The change looks good to me, assuming the tests pass.

I assume none of our automated tests use this because of the Hadoop distribution dependency. Have you confirmed that it works manually as expected?

**Review by mrry — APPROVED** [review]

**Comment by guschmue** [comment]
yes, I tested this against hadoop 2.6.4 running on a linux box. 
Tried to read files for training as well as save() / restore() - worked all ok (actually cool that gfile works).
The biggest hurdle is that you need the hadoop binaries on the windows side and there is no good binary distribution.

File: tensorflow/core/platform/hadoop/hadoop_file_system.cc [hunk 1]
```diff
@@ -27,6 +27,12 @@ limitations under the License.
 #include "tensorflow/core/platform/posix/error.h"
 #include "third_party/hadoop/hdfs.h"
 
+#if defined(PLATFORM_WINDOWS)
+#define LIBHDFS_DSO "hdfs.dll"
```
Comment (jhseu) [inline tensorflow/core/platform/hadoop/hadoop_file_system.cc]: Nit: mind changing these to `const char* kLibHdfsDso = "..."`?

**Review by jhseu — CHANGES_REQUESTED** [review]

File: tensorflow/core/platform/hadoop/hadoop_file_system.cc (hunk 1 above)
Comment (guschmue) [inline tensorflow/core/platform/hadoop/hadoop_file_system.cc]: sure, will do

**Review by jhseu — APPROVED** [review]

**Comment by caisq** [comment]
@tensorflow-jenkins test this please

**Comment by caisq** [comment]
The test timeout in "Linux CPU Tests (Python 3)" seems to be an unrelated flake. Merging the PR now.

**Comment by caisq** [comment]
PR merged. Thanks, @guschmue and reviewers.

**Comment by snnn** [comment]
Some notes for windows users:
1. jvm.dll need be in PATH dirs.
set PATH=%PATH%;%JAVA_HOME%\jre\bin\server

2.  Wildcards in Classpath do not work. This is a limitation of JNI. Please set classpath in this way:
hadoop classpath --jar c.jar
set CLASSPATH=c.jar

**Comment by guschmue** [comment]
I think this is very helpful for people trying to setup hdfs on windows. Let me see if I can add this to the hdfs howto.

**Comment by snnn** [comment]
@guschmue Sure. Please do it.

**Pull request merged**